projects/sandbox/datagen/datagen/utils/priors.py

```diff
@@ -1,109 +1,131 @@
+    param_max = np.max(param_values)
+    if spacing == "lin":
+        bins = np.linspace(param_min, param_max, grid_size + 1)
+        grid = (bins[:-1] + bins[1:]) / 2
+    elif spacing == "log":
+        min_exp = np.log10(param_min)
+        max_exp = np.log10(param_max)
+        bins = np.logspace(min_exp, max_exp, grid_size + 1)
+        grid = np.sqrt(bins[:-1] * bins[1:])
+    else:
+        raise ValueError("Spacing must be either 'lin' or 'log'")
+
+    pdf, _ = np.histogram(param_values, bins, density=True)
+
+    return grid, pdf
+
+
+def read_priors_from_file(event_file: Path, *parameters: str) -> BBHPriorDict:
+    prior = BBHPriorDict()
+    with h5py.File(event_file, "r") as f:
+        events = f["events"]
+        events = events[(events["mass_1"] > 5) & (events["mass_2"] > 5)]
+        field_names = parameters or events.dtype.names
+        for name in field_names:
+            # We'll specify mass_2 via mass_ratio at sampling time
+            if name == "mass_2":
+                continue
+            grid, pdf = pdf_from_events(events[name])
+            prior[name] = Interped(grid, pdf, np.min(grid), np.max(grid))
+
+    return prior
+
+
+def assign_prior_names(prior: BBHPriorDict) -> BBHPriorDict:
+    for key in prior.keys():
+        if hasattr(prior[key], "name"):
+            print("check")
+            prior[key].name = key
+    return prior
+
+
+def uniform_extrinsic():
+    prior = BBHPriorDict()
+    prior["dec"] = Cosine()
+    prior["ra"] = Uniform(0, 2 * np.pi)
+    prior["theta_jn"] = 0
+    prior["phase"] = 0
+    prior = assign_prior_names(prior)
+
+    return prior
+
+
+def nonspin_bbh():
+    prior = uniform_extrinsic()
+    prior["mass_1"] = Uniform(5, 100, unit=msun)
+    prior["mass_2"] = Uniform(5, 100, unit=msun)
+    prior["mass_ratio"] = Constraint(0.2, 5)
+    prior["luminosity_distance"] = UniformSourceFrame(
+        100, 3000, unit=mpc, name="luminosity_distance"
+    )
+    prior["psi"] = 0
+    prior["a_1"] = 0
+    prior["a_2"] = 0
+    prior["tilt_1"] = 0
+    prior["tilt_2"] = 0
+    prior["phi_12"] = 0
+    prior["phi_jl"] = 0
+    prior = assign_prior_names(prior)
+
+    return prior
+
+
+def end_o3_ratesandpops():
+    prior = uniform_extrinsic()
+    prior["mass_1"] = PowerLaw(alpha=-2.35, minimum=2, maximum=100, unit=msun)
+    prior["mass_2"] = PowerLaw(alpha=1, minimum=2, maximum=100, unit=msun)
+    prior["mass_ratio"] = Constraint(0.02, 1)
+    prior["luminosity_distance"] = UniformComovingVolume(
+        100, 15000, unit=mpc, name="luminosity_distance"
+    )
+    prior["psi"] = 0
+    prior["a_1"] = Uniform(0, 0.998)
+    prior["a_2"] = Uniform(0, 0.998)
+    prior["tilt_1"] = Sine(unit=rad)
+    prior["tilt_2"] = Sine(unit=rad)
+    prior["phi_12"] = Uniform(0, 2 * np.pi)
+    prior["phi_jl"] = 0
+    prior = assign_prior_names(prior)
+
+    return prior
+
+
+def power_law_dip_break():
+    prior = uniform_extrinsic()
+    event_file = "/home/william.benoit/\
```

Comment:
Is there a way to do this without hard-coding? How would this look with this as an argument?

Reply:
I'm open to ideas, but I do think having the file name as an argument would be overkill. I would rather each prior function correspond to a specific set of priors than have a generic `read_from_file_prior()`, just for the sake of clarity. Like, this prior corresponds to this file, and that's all it does every time. This way, anyone who wants to use a particular prior doesn't have to wonder which file they need.

Reply:
Disregard, just seeing the discussion about this below